throttle/contract_tx_limiter_middleware.go

```diff
@@ -127,10 +126,27 @@ func NewContractTxLimiterMiddleware(cfg *ContractTxLimiterConfig,
 		if err := proto.Unmarshal(msg.Data, &msgTx); err != nil {
 			return res, errors.Wrapf(err, "unmarshal call tx %v", msg.Data)
 		}
-
 		if msgTx.VmType != vm.VMType_EVM {
 			return next(state, txBytes, isCheckTx)
 		}
+		if txl.inactiveDeployerContracts == nil ||
+			(txl.contractDataLastUpdated+cfg.ContractDataRefreshInterval) < time.Now().Unix() {
+			ctx, err := createUserDeployerWhitelistCtx(state)
+			if err != nil {
+				return res, errors.Wrap(err, "throttle: context creation")
+			}
+			inactiveDeployerContracts, err := udw.GetInactiveDeployerContracts(ctx)
```

Comment:
Why are all the contracts being loaded twice when they could just be loaded once in `GetContractTierMapping` and split into active/inactive all in one go?

Reply:
Only deployer list is being parsed twice, but contracts are being loaded/parsed only once, which is not expensive. Shouldn't we keep functions atomic in this case because otherwise we have to check if feature is enabled which will make the function unnecessarily complex.

Reply:
Each `UserDeployerState` contains a list of contracts deployed by that account, and you're loading every single `UserDeployerState` that was ever created, all that unmarshalling could add up to cause a noticeable hickup every time the caches are reloaded.

Reply:
ok.. changes done